framework/wazuh/core/batcher/mux_demux.py

```diff
@@ -0,0 +1,243 @@
+import os
+import logging
+import signal
+from multiprocessing import Queue, Process, Event
+from multiprocessing.managers import DictProxy, SyncManager
+from typing import Optional, Any
+
+
+logger = logging.getLogger('wazuh-comms-api')
+
+
+class Message:
+    """Message for the MuxDemuxQueue with an associated unique identifier.
+
+    Parameters
+    ----------
+    uid : int
+        Unique identifier for the message.
+    msg : dict
+        Message content as a dictionary.
+    """
+    def __init__(self, uid: int, msg: dict):
+        self.uid = uid
+        self.msg = msg
+
+
+class MuxDemuxQueue:
+    """Queue for managing messages between mux and demux components.
+
+    Parameters
+    ----------
+    proxy_dict : DictProxy
+        Dictionary-like proxy for managing responses.
+    mux_queue : Queue
+        Queue for multiplexing messages.
+    demux_queue : Queue
+        Queue for demultiplexing messages.
+    """
+    def __init__(self, proxy_dict: DictProxy, mux_queue: Queue, demux_queue: Queue):
+        self.responses = proxy_dict
+        self.mux_queue = mux_queue
+        self.demux_queue = demux_queue
+
+    def send_to_mux(self, uid: int, msg: dict) -> int:
+        """Put a message into the mux queue with an associated unique identifier.
+
+        Parameters
+        ----------
+        uid : int
+            Unique identifier of the message.
+        msg : dict
+            Message content to be put into the mux queue.
+
+        Returns
+        -------
+        int
+            Unique identifier of the message.
+        """
+        msg = Message(uid=uid, msg=msg)
+        self.mux_queue.put(msg)
+        return uid
```

Comment:
Why does it return the same UID that it receives? We could make the method just to put the message and make the `BatcherClient.send_event` method return the generated id to make the code simpler.

Reply:
Fixed in b5f17053c4d08ff090ec0208d6fb8b9b971300fd.